lib/Listener/SpamReportListener.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2021 Anna Larch <anna@nextcloud.com>
+ *
+ * @author Anna Larch <anna@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Mail\Listener;
+
+use OCA\Mail\Events\MessageFlaggedEvent;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\Service\AntiSpamService;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use Psr\Log\LoggerInterface;
+
+class SpamReportListener implements IEventListener {
+
+	/** @var LoggerInterface */
+	private $logger;
+
+	/** @var AntiSpamService */
+	private $antiSpamService;
+
+	public function __construct(LoggerInterface $logger,
+								AntiSpamService $antiSpamService) {
+		$this->logger = $logger;
+		$this->antiSpamService = $antiSpamService;
+	}
+
+	public function handle(Event $event): void {
+		if (!$event instanceof MessageFlaggedEvent || $event->getFlag() !== \Horde_Imap_Client::FLAG_JUNK) {
```

Comment:
![image](https://user-images.githubusercontent.com/3902676/128852758-1545071e-565f-47b3-a413-af1a690517a3.png)

![image](https://user-images.githubusercontent.com/3902676/128853020-bbc1e688-7873-49af-a18d-48c198e092ab.png)

Hmm. Is this broken on my end? Or the check might be wrong as `$junk` is imap but `junk` our internal value (passed from the frontend to the backend and forwarded to the event).